### PR TITLE
Add mining test

### PR DESCRIPTION
**Pull request by willrnch**
## Motivation

-  Test the mining feature against a local chain with and without tower
- Test the validator onboarding against a local chain
- Make it possible to process the proof backlog with tower even if the first proof hasn't been sent yet

### CODE REVIEW

File: ol/smoke-tests/src/tests.rs [hunk 1]
```diff
@@ -66,5 +287,168 @@ async fn ol_test_basic_restartability() {
...
+            assert_eq!(err.message, "Server error: could not get tower state");
+        },
+        _ => {
+            panic!("miner state for new account shouldn't exists");
+        },
+    }
+
+    // Proof #0
+    let proof_path = test_environment.metadata.workspace_root.join(
+        format!(
```
Comment (0o-de-lally) [inline ol/smoke-tests/src/tests.rs]: This looks good.
I'll just point out we have some helpers for this kind of thing in:
ol/types/src/fixtures.rs

File: ol/smoke-tests/src/tests.rs [hunk 2]
```diff
@@ -1,15 +1,236 @@
...
+            bail!(
+                "Can't find binary {} at expected path {:?}",
+                name,
+                bin_path
+            );
+        }
+        Ok(bin_path)
+    }
+
+    fn cargo_build(&mut self, package: &str) -> Result<()> {
```
Comment (0o-de-lally) [inline ol/smoke-tests/src/tests.rs]: For these kinds of things, using Command. You can just run in rust debug mode:

`.args["cargo", "run", "-p", "tower"] ` etc..

Then you don't need to pre-build

**Review by 0xzoz — APPROVED** [review]

**Comment by 0xzoz** [comment]
Nice cleanup and tests @willrnch ! great work

**Comment by 0o-de-lally** [comment]
@willrnch Can you rebase this. Also please check that the "tools" tests are passing on your machine. See what they are in /,github/workflows/ci

**Pull request merged**